### DISCUSSION TITLE
/generate-open-graph-preview-image-code-front-matter/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#575] [Comments]
Generate open graph preview image in Code with Front Matter
        
[https://www.eliostruyf.com/generate-open-graph-preview-image-code-front-matter/](https://www.eliostruyf.com/generate-open-graph-preview-image-code-front-matter/)

**Comment by comment-migrator** [comment]
```header
Originally posted by SchwarzDeveloping on 2021-10-21 08:39
```

Special thanks for the Example. As update-tip to you script, you can use the content param for replacements with nodeHtmlToImage:

``    const html = `                ``

`{{title}}`
===========

``    {{subtitle}}         `;   nodeHtmlToImage({   output: 'example.png,   html: html,   content: {   'title': "Hallo World!",   'subtitle': "It works!"   }   }).then(() => {console.log(`Succesfully`)}).catch(e => console.log(e?.message || e));    ``

**Reply by comment-migrator** [reply]
```header
Originally posted on 2021-10-21 09:18
```

Thank you for sharing this Andreas!

**Comment by comment-migrator** [comment]
```header
Originally posted by Kristof on 2022-02-17 16:50
```

Thank you for this script. How do I use this generated image as preview image in the dashboard?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2022-02-17 18:22
```

By default, Front Matter uses the **preview** field in order to show the preview image. If you want to use another field, you can do that by creating a content type, and for your **image** type field set the property **isPreviewImage** to **true**.

[https://frontmatter.codes/d...](https://frontmatter.codes/docs/content-types#image-fields "https://frontmatter.codes/docs/content-types#image-fields")

**Reply by comment-migrator** [reply]
```header
Originally posted by Kristof on 2022-02-18 09:24
```

This is what I thought too, but it seems that it doesn't work with the new subfields, regarding to [https://github.com/estruyf/...](https://github.com/estruyf/vscode-front-matter/issues/197 "https://github.com/estruyf/vscode-front-matter/issues/197").

I tried among others (see GitHub issue) following taxonomy:

\- fields  
...  
\- title: photograph  
\- name: photograph  
\- type: fields  
...  
\- title: Social Media  
\- name: socialmedia  
\- type: image  
\- isPreviewImage: true

But the image doesn't show up in the dashboard.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2022-02-18 18:38
```

Hi Kristof,

This will require an enhancement to the extension. Would you be so kind to share your fields configuration? Created an issue for it here: [https://github.com/estruyf/...](https://github.com/estruyf/vscode-front-matter/issues/257 "https://github.com/estruyf/vscode-front-matter/issues/257")

Elio

**Reply by comment-migrator** [reply]
```header
Originally posted by Kristof on 2022-02-18 19:11
```

Yes, for sure :D

Thanx Elio

**Comment by mayankguptadotcom** [comment]
@estruyf - 

Not a programmer as such. i'm trying to use this on Windows. I do have WSL2 as well although i'm using the project from command prompt itself.
here's my project info -

1. It's hugo based CMS
2. i'm using Reveal Hugo theme
3. I've installed nvm on windows and have installed node v14.17.4
4. I've added `./scripts/social-img.js` file
5. I've also added the code block in `.vscode/settings.json` file

```  "frontMatter.custom.scripts": [{
        "title": "Generate social image",
        "script": "./scripts/social-img.js",
        "nodeBin": "~/Users/mayan/AppData/Roaming/nvm/v14.17.4/node64.exe"
      }]
}
```
Now, I'm not sure if the nodeBin path is correct. I have my doubts about this. I don't think `~` would work in windows.. will it?
When I run the generate social image, i'm getting the following error -

```
Front Matter CMS: Generate social image: Command failed: /Users/mayan/AppData/Roaming/nvm/v14.17.4/node64.exe "c:\Users\mayan\Documents\projects\my-presentation\scripts\social-img.js" "c:\Users\mayan\Documents\projects\my-presentation" "c:\Users\mayan\Documents\projects\my-presentation\content\second\_index.md" "{\"title\":\"Presentations By @mayank\",\"description\":\"Exploring the world of Digital & Agile Transformations, Digital Experience Platforms & Content management systems (CMS)\",\"author\":\"Mayank Gupta\",\"outputs\":[\"Reveal\"]}"
undefined:1
{"title":"Presentations By @mayank","description":"Exploring the world of Digital 

SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (c:\Users\mayan\Documents\projects\my-presentation\scripts\social-img.js:47:62)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47
'Agile' is not recognized as an internal or external command,
operable program or batch file.
'Content' is not recognized as an internal or external command,
operable program or batch file.

```

any suggestions on how I can make this work?

**Reply by estruyf** [reply]
Hi @mayankguptadotcom,

I believe the issue is related to the JSON parsing. Since I wrote the article, I've made a new dependency available to make it easier to write scripts like this. The extension is: `@frontmatter/extensibility`, and more information can be found in the documentation: https://frontmatter.codes/docs/custom-actions#creating-a-content-script

The article just got updated with the new script as well. Additional actions:

- Install the `@frontmatter/extensibility` dependency: `npm i @frontmatter/extensibility`
- Rename the `social-img.js` file to `social-img.mjs`
- Copy the new script content
- Update the `social-img.js` script reference in your `frontMatter.custom.scripts` setting to `social-img.mjs`

Let me know if those changes fixed it.
